File: ex04/ex04_02/ex04_02.js
import promptSync from 'prompt-sync';
const prompt = promptSync();

function main() {
  let year = 0;

  console.log("Enter year: ");
  year = parseInt(prompt());
  if (year < 0) {
    console.log("Invalid year. Exiting.");
    return 1;
  }
  if ((year % 4 == 0 && year % 100 != 0) || year % 400 == 0) {
    console.log("Year " + year + " is a leap year.");
  } 
  else {
     console.log("Year " + year + " is a common year.");
  }
  return 0;
}
main();

File: ex04/ex04_02/ex04_02.py
def main():
  year = 0

  print("Enter year: ")
  year = int(input())
  if year < 0:
    print("Invalid year. Exiting.")
    return 1
  if (year % 4 == 0 and year % 100 != 0) or year % 400 == 0:
    print("Year " + str(year) + " is a leap year.")
  else:
    print("Year " + str(year) + " is a common year.")
  return 0

if __name__ == "__main__":
  main()